classes/Cookie.php

```diff
@@ -250,7 +250,7 @@ public function isLogged($withGuest = false)
         }
 
         /* Customer is valid only if it can be load and if cookie password is the same as database one */
-        if ($this->logged == 1 && $this->id_customer && Validate::isUnsignedId($this->id_customer) && Customer::checkPassword((int) ($this->id_customer), $this->passwd)) {
+        if ($this->logged == true && $this->id_customer && Validate::isUnsignedId($this->id_customer) && Customer::checkPassword((int) ($this->id_customer), $this->passwd)) {
```

Comment:
This change may be dangerous since we can't be sure of the `logged` field type It is not even in the class fields, by the way, we don't know when and where it is created and/or modified

Reply:
My bad it is defined as a `@property bool` so I guess it is modified via the magic getter/setter Since we are not sure of its format then we should indeed keep a loose equality But `true` is fine here